### ISSUE TITLE
Abnormal training results after version 3638c16

### ISSUE THREAD

**Issue by 0ZL0** [neosr-project/neosr#111]
When using version 3638c16, the training results are as expected:

```
# psnr: 15.7619........ Best: 15.7619 @ 1000 iter
# ssim: 0.4419........ Best: 0.4419 @ 1000 iter 
# dists: 0.8014........ Best: 0.8014 @ 1000 iter
```

However, when I try to use versions after 3638c16, I encounter abnormal training data:

```
# psnr: 17.5275........ Best: 17.5275 @ 1000 iter
# ssim: 1.4985........ Best: 1.4985 @ 1000 iter
# dists: 1.2989........ Best: 1.2989 @ 1000 iter
```

The SSIM value is abnormally high, and the DISTS value is abnormally high as well. This issue persists throughout the training and does not improve.

**Environment:**
- neosr version: 3638c16 (working) and later versions (issue)
- Training configuration:
```toml
name = "2xomnisr_ps32_bs8_mssim1.0_adamw_5e-4_0~100000"
model_type = "image"
scale = 2
use_amp = false
bfloat16 = false
fast_matmul = true

[datasets.train]
type = "paired"
dataroot_gt = 'Z:\GT\TRAIN'
dataroot_lq = 'Z:\IN\TRAIN'
patch_size = 32
batch_size = 8
augmentation = [ "none", "mixup", "cutmix", "resizemix"]
aug_prob = [ 0.7, 0.1, 0.1, 0.1 ]

[datasets.val]
name = "val"
type = "paired"
dataroot_gt = 'Z:\GT\VAL'
dataroot_lq = 'Z:\IN\VAL'
[val]
val_freq = 1000
[val.metrics.psnr]
type = "calculate_psnr"
[val.metrics.ssim]
type = "calculate_ssim"
[val.metrics.dists]
type = "calculate_dists"
better = "lower"

[network_g]
type = "omnisr"
upsampling = 2
window_size = 8

[train]
ema = 0.999

[train.optim_g]
type = "adamw"
lr = 5e-4
betas = [ 0.9, 0.99 ]
weight_decay = 0

[train.mssim_opt]
type = "mssim_loss"
loss_weight = 1.0

[logger]
total_iter = 100000
save_checkpoint_freq = 1000
use_tb_logger = true
#save_tb_img = true
print_freq = 100
```
Thank you for your assistance.

**Comment by neosr-project** [comment]
Hi @0ZL0 . Did you try testing before/after using the same config and with manual_seed (deterministic mode)? Because I looked into my commit history after the one you mentioned and can't find anything that would affect metric values. TOPIQ has been added, but other metrics were not modified.
Another possibility is the pytorch version, which has been bumped from 2.5 to 2.6 after that commit. Make sure your driver version has at least CUDA 12.6, since that is required by newer versions of neosr.
One last possibility is the use of `torch.inference_mode()` on the calculate_metric function. However, since none of them should require grads, it should be safe to use it (and indeed I have tested before commiting).

Let me know if you can confirm using deterministic mode (as few as 1k iters is enough, if sha256 hashs match between models that means there was no change).  I can take a deeper look into if you confirm, but so far you're the only person who reported any issue after that commit.

ps: omnisr has been superseeded by ASID, you might want to take a look at that 😃

**Comment by 0ZL0** [comment]
@neosr-project Thank you for the suggestion! I have conducted some further testing. I have tried on a different machine and used three versions, including 3638c16, c67d467 (the next version of 3638c16), and 31c7022 (the latest), with a fixed seed of 1024 in deterministic mode. The conclusion remains the same - 3638c16 is still the last working version in my environment. My system is Windows 11 23H2, with an NVIDIA RTX 2070S and 572.61 drivers. My Python environment is self-configured, not using the PowerShell script provided by neosr. Here are some results of the testing:

![Image](https://github.com/user-attachments/assets/33f54093-33c3-4af3-ae39-dca8fab06ed1)
Image 1 shows the comparison between 3638c16 (left) and the latest 31c7022 (right), where the SSIM and DISTS of 31c7022 look quite problematic.

![Image](https://github.com/user-attachments/assets/7f7f8828-66dc-4650-bbb6-19db10df9f5b)
Image 2 shows the test results of c67d467 (the next version of 3638c16), which also looks concerning.

![Image](https://github.com/user-attachments/assets/983d97f5-3a78-424e-b2d5-fe8144d0b004)
Additionally, the hash of 3638c16 and its next version c67d467 are consistent, but they differ from the latest version 31c7022.

The configuration file I used for this test is as follows:
```toml
name = "2xasid_ps32_bs8_mssim1.0_adamw_5e-4_0~100000"
model_type = "image"
scale = 2
use_amp = false
bfloat16 = false
fast_matmul = true
manual_seed = 1024

[datasets.train]
type = "paired"
dataroot_gt = 'C:\SRDataSets\GT\TRAIN'
dataroot_lq = 'C:\SRDataSets\IN\TRAIN'
patch_size = 32
batch_size = 8
augmentation = [ "none", "mixup", "cutmix", "resizemix"]
aug_prob = [ 0.7, 0.1, 0.1, 0.1 ]

[datasets.val]
name = "val"
type = "paired"
dataroot_gt = 'C:\SRDataSets\GT\VAL'
dataroot_lq = 'C:\SRDataSets\IN\VAL'
[val]
val_freq = 1000
[val.metrics.psnr]
type = "calculate_psnr"
[val.metrics.ssim]
type = "calculate_ssim"
[val.metrics.dists]
type = "calculate_dists"
better = "lower"

[network_g]
type = "asid"
#type = "asid_d8"
#drop = 0.5

[train]
ema = 0.999

[train.optim_g]
type = "adamw"
lr = 5e-4
betas = [ 0.9, 0.99 ]
weight_decay = 0

[train.mssim_opt]
type = "mssim_loss"
loss_weight = 1.0

[logger]
total_iter = 100000
save_checkpoint_freq = 1000
use_tb_logger = true
#save_tb_img = true
print_freq = 100
```

**Comment by neosr-project** [comment]
Interesting. Thanks for testing it @0ZL0 ! I will investigate and find the cause of it (will post it here once found) 👍

**Comment by neosr-project** [comment]
Hi @0ZL0 , sorry for taking so long to debug this. I found the issue (and fixed it on [`75557f4`](https://github.com/neosr-project/neosr/commit/75557f40fee7a16a5434933364a9bb2d3711e877)), it was a stupid mistake from my part. Thanks a lot for reporting and feel free to open new issues 👍
